maxAreaOfIsland.js

```diff
@@ -0,0 +1,126 @@
+/*
+You are given an m x n binary matrix grid. An island is a group of 1's (representing land) connected 4-directionally (horizontal or vertical.) 
+You may assume all four edges of the grid are surrounded by water.
+
+The area of an island is the number of cells with a value 1 in the island.
+
+Return the maximum area of an island in grid. If there is no island, return 0.
+*/
+
+//    0  1  2  3  4  5  6  7  8  9  10 11 12
+// 0 [0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0],
+// 1 [0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0],
+// 2 [0, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0],
+// 3 [0, 1, 0, 0, 1, 1, 0, 0, 1, 0, 1, 0, 0],
+// 4 [0, 1, 0, 0, 1, 1, 0, 0, 1, 1, 1, 0, 0],
+// 5 [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0],
+// 6 [0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0],
+// 7 [0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0],
+
+const { Queue } = require('./utils/queue');
+
+const directions = [
+    [-1, 0],
+    [0, -1],
+    [0, 1],
+    [1, 0],
+];
+
+const ISLAND = 1;
+const WATER = 0;
+
+const getRowColString = (row, col) => `${row}, ${col}`;
+
+const isBound = (newRow, newCol, grid) => {
```

Comment:
This function name makes it sound like it'll take in a positive and tell me if that position is a boundary. That's not what it does. Something like `isInBounds` or `isWithinBounds` would be clearer.